Services/WebToken/AccessTokenLoader.php

```diff
@@ -0,0 +1,166 @@
+     * @param string[] $jwsHeaderChecker
+     * @param string[] $mandatoryClaims
+     * @param string[] $signatureAlgorithms
+     * @param string $signatureKeyset
+     * @param string[] $jweHeaderChecker
+     * @param string[] $keyEncryptionAlgorithms
+     * @param string[] $contentEncryptionAlgorithms
+     * @param string|null $encryptionKeyset
+     */
+    public function __construct(
+        JWSLoaderFactory            $jwsLoaderFactory,
+        ?JWELoaderFactory           $jweLoaderFactory,
+        ClaimCheckerManagerFactory  $claimCheckerManagerFactory,
+        array                       $claimChecker,
+        array                       $jwsHeaderChecker,
+        array                       $mandatoryClaims,
+        array                       $signatureAlgorithms,
+        string                      $signatureKeyset,
+        ?bool                       $continueOnDecryptionFailure,
+        ?array                      $jweHeaderChecker,
+        ?array                      $keyEncryptionAlgorithms,
+        ?array                      $contentEncryptionAlgorithms,
+        ?string                     $encryptionKeyset
+    ) {
+        $this->jwsLoader = $jwsLoaderFactory->create(['jws_compact'], $signatureAlgorithms, $jwsHeaderChecker);
+        if ($jweLoaderFactory !== null && $keyEncryptionAlgorithms !== null && $contentEncryptionAlgorithms !== null && $jweHeaderChecker !== null) {
+            $this->jweLoader = $jweLoaderFactory->create(['jwe_compact'], $keyEncryptionAlgorithms, $contentEncryptionAlgorithms, [], $jweHeaderChecker);
+            $this->continueOnDecryptionFailure = $continueOnDecryptionFailure;
+        }
+        $this->signatureKeyset = JWKSet::createFromJson($signatureKeyset);
+        $this->encryptionKeyset = $encryptionKeyset === null ? null : JWKSet::createFromJson($encryptionKeyset);
+        $this->claimCheckerManager = $claimCheckerManagerFactory->create($claimChecker);
+        $this->mandatoryClaims = $mandatoryClaims;
+    }
+
+    public function load(string $token): array
+    {
+        $token = $this->loadJWE($token);
+        $data = $this->loadJWS($token);
+        try {
+            $this->claimCheckerManager->check($data, $this->mandatoryClaims);
+        } catch (MissingMandatoryClaimException|InvalidClaimException $e) {
+            throw new JWTDecodeFailureException(JWTDecodeFailureException::INVALID_TOKEN, $e->getMessage(), $e, $data);
+        } catch (\Throwable $e) {
+            throw new JWTDecodeFailureException(JWTDecodeFailureException::INVALID_TOKEN, 'Unable to load the token', $e, $data);
+        }
+
+        return $data;
+    }
+
+    /**
+     * @return array<string, mixed>
+     */
+    private function loadJWS(string $token): array
+    {
+        $payload = null;
+        $data = null;
+        $signature = null;
+        try {
+            $jws = $this->jwsLoader->loadAndVerifyWithKeySet($token, $this->signatureKeyset, $signature);
```

Comment:
could we find a way to take the configured `clock_skew` into account for time-based claims? If not, we should mention that limitation in the docs as for the token split feature.

Reply:
It's ok if it's for another PR, people can keep using with the lcobucci integration if they have a strong need for it until we implement that feature.

Reply:
Addressed through https://github.com/lexik/LexikJWTAuthenticationBundle/pull/1041/commits/c9a33d79802a8dc62cfbe8efde433e1a63f4f3a4